src/plugins/intel_cpu/src/nodes/executors/acl/acl_reduce.cpp

```diff
@@ -38,22 +38,32 @@ bool AclReduceExecutor::init(const ReduceAttrs& reduceAttrs,
 
     auto srcDims = srcDescs[0]->getShape().getStaticDims();
     auto dstDims = dstDescs[0]->getShape().getStaticDims();
+    auto srcShape = shapeCast(srcDims);
+    auto dstShape = shapeCast(dstDims);
+    if (srcDescs[0]->hasLayoutType(LayoutType::nspc) && dstDescs[0]->hasLayoutType(LayoutType::nspc)) {
+        changeLayoutToNH_C({&srcShape, &dstShape});
+    }
 
-    TensorInfo srcTensorInfo = TensorInfo(shapeCast(srcDims), 1,
+    TensorInfo srcTensorInfo = TensorInfo(srcShape, 1,
     precisionToAclDataType(srcDescs[0]->getPrecision()), getAclDataLayoutByMemoryDesc(srcDescs[0]));
-    TensorInfo dstTensorInfo = TensorInfo(shapeCast(dstDims), 1,
+    TensorInfo dstTensorInfo = TensorInfo(dstShape, 1,
     precisionToAclDataType(dstDescs[0]->getPrecision()), getAclDataLayoutByMemoryDesc(dstDescs[0]));
 
     srcTensor.allocator()->init(srcTensorInfo);
     dstTensor.allocator()->init(dstTensorInfo);
 
     std::function<std::unique_ptr<IFunction>(void)> exec_func;
+    std::vector<int> castedAxes;
+    for (size_t i = 0; i < reduceAttrs.axes.size(); ++i) {
+        int axis = axisCast(reduceAttrs.axes[i], srcDims.size(), srcDescs[0]->hasLayoutType(LayoutType::nspc) ? NHWC_TO_NCHW : NO_LAYOUT_CONVERSION);
+        if (axis == -1) return false;
```

Comment:
Looks like a sanity check, but why do we check only `-1`, actually any values outside `[-r, r-1]`, where r - is the rank, should be considered invalid.

Reply:
Updated the check. It checks that layout conversion required for `nspc` done successfully.
Unsuccessful return code returned by `axisCast` if layout conversion is enabled is `-1`.

Reply:
Please note that `srcDescs[0]->hasLayoutType(LayoutType::nspc)` is not that cheap, so I would suggest store its result to save one extra call:
```cpp
    const bool is_nspc = srcDescs[0]->hasLayoutType(LayoutType::nspc);
    int axis = axisCast(reduceAttrs.axes[i], srcDims.size(), is_nspc ? NHWC_TO_NCHW : NO_LAYOUT_CONVERSION);
    if (is_nspc && axis == -1) return false;
```

Reply:
Fixed